apollo-federation/src/sources/connect/json_selection/js_literal.rs

```diff
@@ -0,0 +1,185 @@
+// A JSLiteral is similar to a JSON value (or serde_json::Value), with the
+// addition of PathSelection as a possible leaf value, so literal expressions
+// passed to -> methods (via MethodArgs) can capture both field and argument
+// values and sub-paths, in addition to constant JSON structures and values.
+
+use apollo_compiler::collections::IndexMap;
+use nom::branch::alt;
+use nom::bytes::complete::tag;
+use nom::character::complete::char;
+use nom::character::complete::one_of;
+use nom::combinator::map;
+use nom::combinator::opt;
+use nom::combinator::recognize;
+use nom::multi::many0;
+use nom::sequence::delimited;
+use nom::sequence::pair;
+use nom::sequence::preceded;
+use nom::sequence::tuple;
+use nom::IResult;
+use serde::ser::SerializeMap;
+use serde::ser::SerializeSeq;
+use serde::Serialize;
+use serde_json_bytes::Value as JSON;
+
+use super::helpers::spaces_or_comments;
+use super::parser::parse_string_literal;
+use super::parser::Key;
+use super::parser::PathSelection;
+
+#[derive(Debug, PartialEq, Clone)]
+pub enum JSLiteral {
+    String(String),
+    Number(String),
```

Comment:
Can we store `serde_json::Number` here instead of allocating a new string? Then we can defer to `serde_json::Number::from_str` during parsing to make sure that a valid literal number in JSONSelection is always a valid JSON number at execution.

Reply:
Yes, that works nicely! https://github.com/apollographql/router/pull/5762/commits/ab8bcd76b7ffb88354ea17dac92ffb8e1631855b